src/components/body/body-row.component.ts

```diff
@@ -129,7 +138,14 @@ export class DataTableBodyRowComponent implements DoCheck {
     };
 
     if (group === 'left') {
-      translateXY(styles, offsetX, 0);
+
+      if (this.groupColumns){
+        //offset x position in case there are group columns
+        translateXY(styles, offsetX-3, 0);
```

Comment:
This is the comment I am adding to the code as well:
"
The grouping of rows add some elements to ngx-datatable which total 3 pixels on the x axis which needs to be offset to maintain the header columns aligned with the data columns.
"